presto-main/src/main/java/com/facebook/presto/execution/QueryInfo.java

```diff
@@ -60,6 +60,8 @@
     private final String updateType;
     private final Optional<StageInfo> outputStage;
     private final FailureInfo failureInfo;
+    private final Optional<String> failureTask;
```

Comment:
Yes, I agree that we should put these two fields into `FailureInfo` if possible.

But the problem is that `FailureInfo` is from `ExecutionFailureInfo` which is used in lots of places. Putting these two fields into `FailureInfo` might require lots of changes in related places, which can be much more complex.

@raghavsethi What do you think?